folsom/src/main/java/com/spotify/folsom/client/ascii/AsciiMemcacheDecoder.java

```diff
@@ -216,7 +216,17 @@ protected void decode(final ChannelHandlerContext ctx, final ByteBuf buf, final
           out.add(AsciiResponse.NOT_STORED);
           return;
         } else {
-          throw fail();
+          final String lineStr = toString(line);
+          switch (lineStr) {
+            case "SERVER_ERROR object too large for cache":
+              out.add(AsciiResponse.VALUE_TOO_LARGE);
+              return;
+            case "SERVER_ERROR out of memory storing object":
+              out.add(AsciiResponse.OUT_OF_MEMORY);
+              return;
+            default:
+              throw new IOException("Unexpected line: " + lineStr);
```

Comment:
Perhaps make the error message a bit more descriptive.

Reply:
This is the same as it was before this change though, so I will leave it alone